packages/2d/src/lib/components/Latex.ts

```diff
@@ -246,4 +262,175 @@ export class Latex extends SVGNode {
+      animations.push(
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].end(1, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].fill(this.color, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].lineWidth(0, duration, timingFunction),
+        ),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *unwrite(time: number, timingFunction?: TimingFunction) {
+    const paths = this.getPaths();
+    const duration = time / paths.length;
+    const animations: ThreadGenerator[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      paths[i].lineWidth(0).stroke(this.color).start(0);
+      animations.push(
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].fill(null, duration, timingFunction),
+        ),
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].lineWidth(40, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].start(1, duration, timingFunction),
+        ),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *edit(tex: string, time: number) {
+    yield* this.tex(math(tex), time);
+  }
+
+  public *morph(ptx: Latex, time: number) {
+    const paths = this.getPaths();
+    const targets = ptx.getPaths();
+    const animations: ThreadGenerator[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      paths[i].stroke(this.fill());
+
+      animations.push(
+        paths[i].fill(null, time * BUFFER),
+        paths[i].lineWidth(40, time * BUFFER),
+        delay(
+          time * BUFFER,
+          paths[i].data(targets[i].data(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(
+          time * BUFFER,
+          paths[i].position(targets[i].position(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(
+          time * BUFFER,
+          paths[i].scale(targets[i].scale(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(time * (1 - BUFFER), paths[i].fill(paths[i].stroke(), time / 2)),
+        delay(time * (1 - BUFFER), paths[i].lineWidth(0, time / 2)),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *map(ptx: Latex, mapping: number[][], time: number) {
+    let paths = this.getPaths();
+    const len = paths.length;
+    const targets = ptx.getPaths();
+    const animations: ThreadGenerator[] = [];
+    const taken = targets.map(() => false);
+    const extras: number[] = [];
+
+    for (let i = 0; i < len; i++) {
+      if (!mapping[i].length) {
+        animations.push(
+          delay(time * BUFFER, paths[i].opacity(0, time * (1 - 2 * BUFFER))),
+        );
+        extras.push(i);
+        continue;
+      }
+
+      for (let j = 0; j < mapping[i].length; j++) {
+        let index = i;
+
+        if (j > 0) {
+          const subToPihedron = this.childAs(0)?.childAs(0);
```

Comment:
Cute, but I'm not a fan of self-promo like this. I get it when it's in a resource on the discord server or whatever, but ideally not here.